### PR TITLE
[NUI-Text] update to include text encoding

### PR DESCRIPTION
**Pull request by Ali-Alzyoud**
### Change Description ###
add description of how to use encoded characters and using encoded entities in MarkupText

Describe your changes here.

For instance,
 - Fix broken links of native application docs.
 -  ...

### Bugs Fixed ###

Provide links to bugs here

For instance,

 - Issue #265

### API Changes ###

In case of ACR, put the link of ACR

For instance,

 - ACR: ACR-1120

### CODE REVIEW

File: docs/application/dotnet/guides/nui/text.md [hunk 1]
```diff
@@ -150,6 +150,17 @@ To align the text in a text label:
...
+<a name="encoding"></a>
+### Text property can accept UTF16 or UTF32 character encoding
+
+```
+TextLabel title = new TextLabel();
+title.Text = "\U0001f601";      //UTF-32
+title.Text += "\ud83d\ude01";   //UTF-16
+```
+
+
```
Comment (surbhig12) [inline docs/application/dotnet/guides/nui/text.md:162]: Please remove extra line

File: docs/application/dotnet/guides/nui/text.md [hunk 2]
```diff
@@ -342,6 +353,19 @@ The following markup elements are currently supported:
...
+### Use Markup to represent encoded characters
+
+Since Markup text is not allowed to contain some characters unless they are representing tags such as "<" or ">", to use these characters or represent any unicode character, user should use reserved entities.
+```
+TextLabel label = new TextLabel();
+label.EnableMarkup = true;
+label.Text = "&lt;&#x1f601;&gt;"; //less-than Smileface greater-than
+```
+
+
```
Comment (surbhig12) [inline docs/application/dotnet/guides/nui/text.md:367]: Please remove extra line

File: docs/application/dotnet/guides/nui/text.md [hunk 3]
```diff
@@ -342,6 +353,19 @@ The following markup elements are currently supported:
 
 
 
+<a name="decorations"></a>
+
+### Use Markup to represent encoded characters
+
+Since Markup text is not allowed to contain some characters unless they are representing tags such as "<" or ">", to use these characters or represent any unicode character, user should use reserved entities.
```
Comment (surbhig12) [inline docs/application/dotnet/guides/nui/text.md:360]: Suggest rephrasing to:

Markup text is not allowed to contain some characters unless they are representing tags such as "<" or ">". To use these characters or represent any unicode character, you must use reserved entities.

Please verify for technical accuracy.

**Review by surbhig12 — COMMENTED** [review]
Reviewed modified content. Please verify.

File: docs/application/dotnet/guides/nui/text.md [hunk 4]
```diff
@@ -150,6 +150,15 @@ To align the text in a text label:
 
 
 
+<a name="encoding"></a>
+### Text property can accept UTF16 or UTF32 character encoding
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/text.md:154]: How about simplifying the title, and adding one simple description?
For example

\### UTF16 or UTF32 character encoding

The text property can accept UTF16 or UTF32 character encoding

**Review by mijin85cho — COMMENTED** [review]
Hello. I also have some comments. Would you please check them?

**Review by hyunjushin — APPROVED** [review]

**Review by surbhig12 — APPROVED** [review]
Thank you for incorporating the comments.

File: docs/application/dotnet/guides/nui/text.md [hunk 5]
```diff
@@ -150,6 +150,16 @@ To align the text in a text label:
 
 
 
+### UTF16 or UTF32 character encoding
+
+The text property can accept UTF16 or UTF32 character encoding
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/text.md:155]: @surbhig12 I think checking the colons before the code block is missed.

```suggestion
The text property can accept UTF16 or UTF32 character encoding:

```

File: docs/application/dotnet/guides/nui/text.md [hunk 6]
```diff
@@ -342,6 +352,15 @@ The following markup elements are currently supported:
 
 
 
+### Use Markup to represent encoded characters
+
+Markup text is not allowed to contain some characters unless they are representing tags such as "<" or ">". To use these characters or represent any unicode character, you must use reserved entities.
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/text.md:357]: @surbhig12 I thought according to the SRI-B style, we need a sentence here.

The following example uses reserved entities:

Would you please check the sentence?

File: docs/application/dotnet/guides/nui/text.md (hunk 5 above)
Comment (surbhig12) [reply docs/application/dotnet/guides/nui/text.md:155]: THank you for pointign this out.

File: docs/application/dotnet/guides/nui/text.md (hunk 6 above)
Comment (surbhig12) [reply docs/application/dotnet/guides/nui/text.md:357]: Suggestion:

Markup text is not allowed to contain some characters unless they are representing tags such as "<" or ">". To use these characters or represent any unicode character, you must use reserved entities. The following example uses reserved entities:

Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/text.md:357]: @surbhig12 Your suggestion has no big difference with mine, so I hope you to check again it there wasn't a mistake.

Comment (surbhig12) [reply docs/application/dotnet/guides/nui/text.md:357]: @mijin85cho, the intention was not to change the sentence completely. I agree with your suggestion of "The following example uses reserved entities:", therefore I added it as a followign sentence.

Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/text.md:357]: Thanks for the reply, @surbhig12 

@Ali-Alzyoud as there were some missed points, would you please update them also? After that I'll ask @surbhig12 to check and approve again to proceed to next step.

File: docs/application/dotnet/guides/nui/text.md [hunk 7]
```diff
@@ -342,6 +352,27 @@ The following markup elements are currently supported:
...
+
+Markup text can include character with UTF32 representation as entities contain decimal or hexadecimal values.
+To represent decimal value, user can use : `&#` + `utf32_decimal_value` + `;`
+To represent hexadecimal value, user can use : `&#x` + `utf32_hexadecimal_value` + `;`
+The following example uses UTF32 entities:
+
+```
+TextLabel label = new TextLabel();
+label.EnableMarkup = true;
+label.Text = "&#9786; &#x263a;"; //smile-face smile-face
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/text.md:373]:
```suggestion
label.Text = "&#9786; &#x263a;"; //smile-face

```

**Comment by mijin85cho** [comment]
Thanks for the update, @Ali-Alzyoud 
@surbhig12 The content is enhanced, so we need to review it again. Would you please check it?

File: docs/application/dotnet/guides/nui/text.md [hunk 8]
```diff
@@ -342,6 +352,27 @@ The following markup elements are currently supported:
 
 
 
+### Use Markup to represent encoded characters
+
+Markup text is not allowed to contain some characters unless they are representing tags or entities such as "<", ">" or "&". To include these characters as a part of the text, user should use reserved entities like `&lt;`, `&gt;` or `&amp;`, The following example uses reserved entities:
```
Comment (surbhig12) [inline docs/application/dotnet/guides/nui/text.md:357]: Suggest rephrasing to:

```suggestion
Markup text is not allowed to contain some characters unless they are representing tags or entities such as "<", ">" or "&". To include these characters as a part of the text, you must use reserved entities such as `&lt;`, `&gt;` , or `&amp;`. The following example uses reserved entities:
```

File: docs/application/dotnet/guides/nui/text.md (hunk 7 above)
Comment (surbhig12) [reply docs/application/dotnet/guides/nui/text.md:373]: Please incorporate this comment.

File: docs/application/dotnet/guides/nui/text.md [hunk 9]
```diff
@@ -342,6 +352,27 @@ The following markup elements are currently supported:
...
+
+```
+TextLabel label = new TextLabel();
+label.EnableMarkup = true;
+label.Text = "&lt;&gt;"; //less-than greater-than
+```
+
+Markup text can include character with UTF32 representation as entities contain decimal or hexadecimal values.
+To represent decimal value, user can use : `&#` + `utf32_decimal_value` + `;`
+To represent hexadecimal value, user can use : `&#x` + `utf32_hexadecimal_value` + `;`
```
Comment (surbhig12) [inline docs/application/dotnet/guides/nui/text.md:367]:
```suggestion
To represent decimal value, you can use: `&#` + `utf32_decimal_value` + `;`
To represent hexadecimal value, you can use: `&#x` + `utf32_hexadecimal_value` + `;`
```

**Review by surbhig12 — COMMENTED** [review]
Minor comments. Please update.

File: docs/application/dotnet/guides/nui/text.md (hunk 8 above)
Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/text.md:357]: @surbhig12 I edited your comment to remove unneeded space.

Comment (surbhig12) [reply docs/application/dotnet/guides/nui/text.md:357]: Thank you

**Review by surbhig12 — APPROVED** [review]
Thank you for incorporating the comments

**Review by mijin85cho — APPROVED** [review]
Thank you for the updates and reviews.

**Pull request merged**